File: dcp-solutions/2024-11-26/solution.go
```go
func merge(intervals []tuple) []tuple {
    // Sort intervals by start time and then end time
    sort.Slice(intervals, func(i, j int) bool {
        if intervals[i].Start < intervals[j].Start {
            return true
        } else if intervals[i].End > intervals[j].End {
            return false
        }
        return intervals[i].Start <= intervals[j].Start && intervals[i].End >= intervals[j].End
    })

    merged := []tuple{intervals[0]}
    for _, interval := range intervals {
        last := merged[len(merged) - 1]
        if interval.Start <= last.End {
            // Merge intervals
            merged[len(merged)-1] = tuple{last.Start, max(last.End, interval.End)}
        } else {
            // Add to the list of merged intervals
            merged = append(merged, interval)
        }
    }

    return merged
}

type tuple struct {
	Start int
	End   int
}
func max(a,b int) int {
    if a > b {
        return a
    }else{
      return  b
    }
}
```

File: dcp-solutions/2024-11-26/solution.c
```c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    int start;
    int end;
} Interval;

Interval* merge(intervals[][]){
    if(!intervals || !*intervals) return NULL;

    // Sort intervals by their start value
    qsort(intervals, sizeof(Interval*) * (*intervals), sizeof(Interval), compare_intervals);

    // Initialize the result with the first interval
    Interval* result = (Interval*)malloc(sizeof(Interval));
    result->start = intervals[0][0];
    result->end = intervals[0][1];

    for(int i=1; i<(*intervals); i++){
        if(intervals[i][0] <= result->end){
            // Merge the current interval to the previous one
            result->end = (result->end > intervals[i][1]) ? result->end : intervals[i][1];
        } else{
            // Add the current interval to the result
            Interval* new_interval = (Interval*)malloc(sizeof(Interval));
            new_interval->start = intervals[i][0];
            new_interval->end = intervals[i][1];

            // Add a copy of the new interval to the end of the result array
            result = (Interval*)realloc(result, (result + 1) * sizeof(Interval));
            memcpy(result + 1, new_interval, sizeof(Interval));
        }
    }

    return result;
}

int compare_intervals(const void* a, const void* b){
    Interval* i1 = (Interval*)a;
    Interval* i2 = (Interval*)b;

    if(i1->start == i2->start) return i1->end - i2->end;
    else return i1->start - i2->start;
}
```